File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = []
        
        nums.sort()

        for i in range(len(nums)):
            copy = nums.copy()
            copy.pop(i)

            start = 0
            end = len(copy)-1

            while start < end:
                l = [nums[i], copy[start], copy[end]]
                l.sort()
                if copy[start] + copy[end] + nums[i] == 0 and l not in result:
                    result.append(l)
                elif copy[start] + copy[end] + nums[i] < 0:
                    start += 1
                else:
                    end -= 1
        
        return result

File: Data Structures & Algorithms/boats-to-save-people/submission-0.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int start = 0;
        int end = people.length - 1;
        int result = 0;
        Arrays.sort(people);

        while (start <= end){
            int remaining = limit -  people[end--];
            if (start <= end && remaining >= people[start]){
                start++;
            }
            result++;
        }
        return result;
    }
}